### PR TITLE
feat(arvp): add deterministic replay scheduler with speed profiles (#1842)

### LINKED ISSUE #1842: [ARVP][SCHEDULER] Implement event-time replay scheduler with deterministic speed profiles
## Summary
Implement an ARVP replay scheduler that controls replay pacing in event time and supports deterministic speed profiles such as 1x, 2x, 5x, 10x, and instant batch mode.

The replay foundation already has a deterministic time context via #1801, but the current `speedup_factor` is only an informational hint and does not yet provide real scheduling behavior. This means the platform can replay deterministically, but not yet as a first-class accelerated validation system.

This slice converts replay speed from a concept into an operational capability.

## Why this slice
The original platform intent explicitly included accelerated validation against historical data. Without a scheduler:
- replay acceleration is not a real product feature yet
- operators cannot run controlled fast-forward validation profiles
- future scenario orchestration has no pacing layer to target
- the platform remains closer to a batch replay tool than to a fast-forward validation engine

## Scope
Add a replay scheduler layer that:
- uses event time rather than wall-clock time as the replay authority
- supports deterministic speed profiles such as:
  - 1x
  - 2x
  - 5x
  - 10x
  - instant / no-wait batch mode
- exposes explicit pacing metadata in run artifacts
- integrates cleanly with the current replay CLI / runner surfaces
- does not change the semantic replay outputs produced by the deterministic foundation
- fail-closes on invalid scheduler configs

## Non-goals
This issue is **not** for:
- DB-backed dataset loading
- scenario pack definitions
- replay-vs-paper comparison
- operator dashboards
- live/paper/Redis service integration
- introducing non-deterministic sleeps, race conditions, or thread-driven timing behavior
- changing business logic in the strategy or execution layers

## Suggested implementation boundaries
Possible surfaces:
- `core/replay/scheduler.py`
- scheduler config added to replay run config
- scheduler metadata embedded into replay artifacts/reports
- CLI wiring in `services/validation/strategy_replay_runner.py`

Suggested behavior:
- derive replay pacing from event timestamps and requested speed profile
- allow instant batch mode for max-throughput offline validation
- preserve deterministic ordering and deterministic outputs regardless of profile
- record which scheduler profile was used for auditability

## Acceptance criteria
- ARVP supports explicit deterministic speed profiles
- Scheduler config is validated fail-closed
- Replaying the same input with the same profile yields identical replay outputs
- Scheduler metadata is captured in run artifacts
- Instant mode is supported for pure offline throughput use cases
- Existing replay foundation behavior is not semantically altered by the scheduler layer

## Suggested tests
- 1x profile validates correctly
- 2x / 5x / 10x profiles validate correctly
- instant mode validates correctly
- invalid speed profile fails closed
- same input + same profile produces identical outputs
- different pacing profiles do not alter business-level replay outcomes
- scheduler metadata is present in produced artifacts

## Definition of done
- event-time scheduler exists
- deterministic speed profiles exist
- scheduler is wired into replay execution surfaces
- artifacts record scheduler provenance
- tests prove deterministic behavior across supported profiles

## Follow-up dependency
This issue prepares:
- ARVP replay job orchestration
- scenario harness
- replay run comparison across scenario packs
- operator-facing fast-forward validation workflows

### PR DESCRIPTION
**Pull request by jannekbuengener**
## Summary
- add new core/replay/scheduler.py for deterministic event-time replay scheduling with validated speed profiles
- add focused scheduler unit tests for profiles, boundaries, determinism, and artifact metadata
- wire the scheduler minimally into strategy_replay_runner with backward-compatible default speedup_profile="instant"
- keep scope tight: no runner orchestration, scenario, or reporting expansion

## Validation
- python -m pytest tests/unit/replay/test_scheduler.py tests/unit/validation/test_strategy_replay_runner.py -q

Closes #1842

### CODE REVIEW

File: services/validation/strategy_replay_runner.py [hunk 1]
```diff
@@ -268,11 +283,60 @@ def _redact_env() -> dict[str, str]:
...
+            fingerprint=spec.fingerprint(),
+            warmup_count=warmup_count,
+            effective_candle_count=len(candles) - warmup_count,
+        )
+        return ReplayScheduler().schedule(
+            dataset_result,
+            SchedulerConfig(profile=config.speedup_profile),
+        ).to_dict()
+    except ValueError as exc:
+        raise ReplayRunnerError(f"scheduler validation failed: {exc}") from exc
```
[the remark this answers is not in the document]
Comment (jannekbuengener) [reply services/validation/strategy_replay_runner.py]: Fixed in `7f294261`. JSON-array candle rows are now validated as objects in `_load_candles`, and `_build_scheduler_metadata` now keeps `TypeError` on the documented fail-closed `ReplayRunnerError` path. Added targeted tests for the malformed-array case and the CLI exit-code-2 mapping.

**Comment by jannekbuengener** [comment]
## Self-Review

**Scope**: Completes the deterministic replay scheduler slice for #1842 and hardens replay input handling for malformed JSON-array candle rows.
**Not touched**: No replay orchestration (#1843), no scenario logic, no contract changes outside the runner/scheduler slice.
**Risk**: low — changes stay inside the replay scheduler/runner surface and add fail-closed validation for malformed inputs.
**Tests**: python -m pytest tests/unit/replay/test_scheduler.py tests/unit/validation/test_strategy_replay_runner.py -q -> 93 passed
**Rollback**: git revert 7f294261
**Evidence**: Required checks green on PR #1859, including ci (Unit/Integration + Lint gesammelt) and policy-gate.
